rskj-core/src/main/java/org/ethereum/datasource/KeyValueDataSource.java

```diff
@@ -140,7 +140,7 @@ static void validateDbKind(DbKind currentDbKind, String databaseDir, boolean dat
         File dir = new File(databaseDir);
         boolean databaseDirExists = dir.exists() && dir.isDirectory();
 
-        if (!databaseDirExists) {
+        if (!databaseDirExists || dir.list().length == 0) {
```

Comment:
`dir.list()` may return null if `databaseDir` is not a directory, causing a NPE.

This is not so important, as it is run just on startup and it would mean a misconfiguration, but I think it is worth fixing it now that we are working on it.

Reply:
`&& dir.isDirectory()` would make this impossible to happen

Reply:
What if it is a regular file?
`dir.exists() == true`
`dir.isDirectory() == false`
`databaseDirExists == false` => `!databaseDirExists == true` and `dir.list().length == 0` will be checked

Reply:
if `!databaseDirExists == true` it ends up there since it is an or conditional

Reply:
Added a validation if `database.dir` is not a folder. Suggested by @Vovchyk

Reply:
> if `!databaseDirExists == true` it ends up there since it is an`OR` conditional

You're right, I don't know what I understood there... 🤯